osu.Game/Rulesets/Mods/ModTimeRamp.cs

```diff
@@ -75,17 +75,24 @@ public virtual void ApplyToBeatmap(IBeatmap beatmap)
             finalRateTime = firstObjectStart + FINAL_RATE_PROGRESS * (lastObjectEnd - firstObjectStart);
         }
 
+        public double ApplyToRate(double time, double rate = 1)
+        {
+            double amount = (time - beginRampTime) / Math.Max(1, finalRateTime - beginRampTime);
+            double ramp = InitialRate.Value + (FinalRate.Value - InitialRate.Value) * Math.Clamp(amount, 0, 1);
+
+            // round the end result to match the bindable SpeedChange's precision, in case this is called externally.
+            return rate * Math.Round(ramp, 2);
```

Comment:
I think this is going to change the runtime behaviour of the mod to be less smooth in the ramping, no? Not sure if it's enough to be audible (and whether it sounds bad) yet, but worth noting.

Reply:
I think that was the case already. The bindable was rounding in its own anyways due to the precision set.

Reply:
Oh. Well then. I plan on testing this out to see how it feels sounds, but should be fine in that case.